### DISCUSSION TITLE
Distributed Hardware Solution?

### DISCUSSION THREAD

**Discussion by NateEaton** [mcguirepr89/BirdNET-Pi#1116] [Q&A]
I'm curious whether anyone involved in the project has considered or developed a distributed architecture in which a separate audio recording-only component, ideally battery or solar powered, could be placed outside in a weather-proof enclosure (perhps based on something like an ESP32 platform) that transmits wirelessly the audio stream to a Raspberry Pi running BirdNET-Pi that's located indoors?

I have a sheltered patio so have considered building a standard BirdNET-Pi configuration in an enclosure that would address condensation (since direct rain won't be an issue on the patio) but with temperatures in the summer up around 100-105 (or higher run under the patio roof), I'm just concerned that no amount of cooling is going to keep a pi4 or even a pi3 running at reasonable temperatures. It would be much better if I could keep the pi inside and have the mic connected wirelessly from outdoors. 

Thoughts?

**Comment by dorsetwurzel** [comment]
Not exactly what you are after, and it uploads data to its own associated website but have a look at this hardware package - this may do what you want but in a different way.

https://www.birdweather.com/

**Comment by pddpauw** [comment]
if you could build a rtsp stream (eg with the esp32), then you should be able to add this audio stream to birdnet, and then you'ready...  The challenge will be in building and hosting that rtsp stream, but sure google can help!

**Comment by PeterMescher** [comment]
I have a Pi0 hosting an RTSP stream of a USB mic, and then that feeds into an LXC container I have on a proxmox server running the containerized version of BirdNet.

I will say that batter/solar would be a tall order; the RTSP config keeps the CPU humming continuously.  That said, it’s not *so* intense, that I’d worry about heat.
